src/types/context.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides conversion methods between types while respecting the context that is the current
+//! [`ProtocolParameters`](bee_block_stardust::protocol::ProtocolParameters).
+
+/// The equivalent to [`TryFrom`] but with an additional context.
+pub trait TryFromWithContext<T>: Sized {
```

Comment:
Maybe a bit too nit-picky, but maybe a slightly better design of this trait (considering how abstract it has been named) would be to make the context an associated type, or if you don't want that...rename the trait to something like `TryFromWithProtocolParameters`.